modules/coin/rest/handlers.go

```diff
@@ -0,0 +1,119 @@
+
+	"github.com/tendermint/basecoin"
+	"github.com/tendermint/basecoin/client/commands"
+	"github.com/tendermint/basecoin/client/commands/proofs"
+	"github.com/tendermint/basecoin/modules/auth"
+	"github.com/tendermint/basecoin/modules/base"
+	"github.com/tendermint/basecoin/modules/coin"
+	"github.com/tendermint/basecoin/modules/fee"
+	"github.com/tendermint/basecoin/modules/nonce"
+	"github.com/tendermint/basecoin/stack"
+	lightclient "github.com/tendermint/light-client"
+	"github.com/tendermint/tmlibs/common"
+)
+
+// SendInput is the request to send an amount from one actor to another.
+// Note: Not using the `validator:""` tags here because SendInput has
+// many fields so it would be nice to figure out all the invalid
+// inputs and report them back to the caller, in one shot.
+type SendInput struct {
+	Fees     *coin.Coin `json:"fees"`
+	Multi    bool       `json:"multi,omitempty"`
+	Sequence uint32     `json:"sequence"`
+
+	To     *basecoin.Actor `json:"to"`
+	From   *basecoin.Actor `json:"from"`
+	Amount coin.Coins      `json:"amount"`
+}
+
+func RegisterHandlers(r *mux.Router) error {
+	r.HandleFunc("/build/send", doSend).Methods("POST")
+	r.HandleFunc("/query/account/{signature}", doQueryAccount).Methods("GET")
+	return nil
+}
+
+// doQueryAccount is the HTTP handlerfunc to query an account
+// It expects a query string with
+func doQueryAccount(w http.ResponseWriter, r *http.Request) {
+	query := mux.Vars(r)
+	signature := query["signature"]
+	actor, err := commands.ParseActor(signature)
+	if err != nil {
+		common.WriteError(w, err)
+		return
+	}
+	actor = coin.ChainAddr(actor)
+	key := stack.PrefixedKey(coin.NameCoin, actor.Bytes())
+	account := new(coin.Account)
+	proof, err := proofs.GetAndParseAppProof(key, account)
+	if lightclient.IsNoDataErr(err) {
+		err := fmt.Errorf("account bytes are empty for address: %q", signature)
+		common.WriteError(w, err)
+		return
+	} else if err != nil {
+		common.WriteError(w, err)
+		return
+	}
+
+	if err := proofs.OutputProof(account, proof.BlockHeight()); err != nil {
+		common.WriteError(w, err)
+		return
+	}
+	common.WriteSuccess(w, account)
+}
+
+func doSend(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	si := new(SendInput)
+	if err := common.ParseAndValidateRequestJSON(r, si); err != nil {
+		common.WriteError(w, err)
+		return
+	}
+
+	var errsList []string
+	if si.From == nil {
+		errsList = append(errsList, `"from" cannot be nil`)
+	}
+	if si.Sequence <= 0 {
+		errsList = append(errsList, `"sequence" must be > 0`)
+	}
+	if si.To == nil {
+		errsList = append(errsList, `"to" cannot be nil`)
+	}
+	if len(si.Amount) == 0 {
+		errsList = append(errsList, `"amount" cannot be empty`)
+	}
+	if len(errsList) > 0 {
+		err := &common.ErrorResponse{
+			Error: strings.Join(errsList, ", "),
+			Code:  406,
+		}
+		common.WriteCode(w, err, 406)
+		return
+	}
+
+	tx := coin.NewSendOneTx(*si.From, *si.To, si.Amount)
+	// fees are optional
+	if si.Fees != nil && !si.Fees.IsZero() {
```

Comment:
So I can extract that into a helper function, but I might have to keep that in modules/coin/rest because client/rest imports modules/coin/rest so that'd be an import cycle.